packages/remix-dev/config.ts

```diff
@@ -162,10 +164,37 @@ export async function readConfig(
 
   let rootDirectory = path.resolve(remixRoot);
   let configFile = path.resolve(rootDirectory, "remix.config.js");
+  let tsConfigFile = path.resolve(rootDirectory, "remix.config.ts");
 
   let appConfig: AppConfig;
   try {
-    appConfig = require(configFile);
+    let configFileExists = fs.existsSync(configFile);
+    let tsConfigFileExists = fs.existsSync(tsConfigFile);
+    if (tsConfigFileExists) {
+      if (tsConfigFileExists && configFileExists) {
+        throw new Error(
+          "Both `remix.config.ts` and `remix.config.js` exist. Please delete either as Remix doesn't know which one to use."
+        );
+      }
+
+      let temporaryFile = path.resolve(os.tmpdir(), "_remix.config.js");
```

Comment:
this would cause problem if you write `path.join(__dirname, './some-path')` in your config, `__dirname` will be the tmpdir.

Reply:
I could also generate a _remix.config.js file in the project root just temporarily, then unlink it immediately. Would that be better?

Reply:
If I do that I might not need the bundle option either

Reply:
you do need the bundle option since you might `import` another ts file in your config

Reply:
I could probably generate the temporary file in the project root instead, then __dirname would still work?

Reply:
maybe check out https://github.com/egoist/bundle-require/blob/main/README.md#how-it-works, it's how Vite loads vite.config.ts

Reply:
Yeah, I might use that instead. Hand rolling my own solution is likely going to cause some problems down the line